### PR TITLE
Enable paging for 64-bit vexpress

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
`Tested-by: Jens Wiklander <jens.wiklander@linaro.org> (Juno AArch64)`

**Comment by jenswi-linaro** [comment]
Adds a fix for https://github.com/OP-TEE/optee_test/pull/216

**Comment by jforissier** [comment]
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (Hikey GP)`

Tested with https://github.com/OP-TEE/optee_os/pull/1592 and https://github.com/OP-TEE/optee_test/pull/216.

File: core/arch/arm/kernel/ssvce_a32.S [hunk 1]
```diff
@@ -55,6 +55,7 @@
 FUNC secure_mmu_unifiedtlbinvall , :
 UNWIND(	.fnstart)
 
+	dsb	/* Ensure visibility of the update to translation table walks */
```
Comment (jenswi-linaro) [inline core/arch/arm/kernel/ssvce_a32.S]: Yes, I'd rather take that in a later patch.

File: core/arch/arm/mm/tee_pager.c [hunk 2]
```diff
@@ -1403,6 +1406,8 @@ static void pager_save_and_release_entry(struct tee_pager_pmem *pmem)
 
 	area_get_entry(pmem->area, pmem->pgidx, NULL, &attr);
 	area_set_entry(pmem->area, pmem->pgidx, 0, 0);
+	/* TODO only invalidate entry touched above */
+	core_tlb_maintenance(TLBINV_UNIFIEDTLB, 0);
```
Comment (jenswi-linaro) [inline core/arch/arm/mm/tee_pager.c]: Yes, I noticed that too. :-)

**Comment by jenswi-linaro** [comment]
Rebased, tags applied. Note that "￼core: pager: invalidate tlb when clearing entry" doesn't have an R-B tag, but I not sure it matters much.

**Comment by jforissier** [comment]
> Note that "￼core: pager: invalidate tlb when clearing entry" doesn't have an R-B tag, but I not sure it matters much

It's OK with me. Could you mention that it fixes xtest 1016, maybe?

**Comment by jenswi-linaro** [comment]
Commit message in "core: pager: invalidate tlb when clearing entry" updated.

**Comment by jforissier** [comment]
Merged, thanks. Travis failure is due to the build exceeding the time limit.

**Comment by ken08liu14** [comment]
Hello Jenswi,
May I ask which platform are you tested ? Seems my platform could not hold even pager part for current.

**Comment by jenswi-linaro** [comment]
QEMU-Virt-ARMv8, Fast Models Foundation model, Juno.
@jforissier has also tested on Hikey with https://github.com/OP-TEE/optee_os/pull/1592 rebased on top
If you look through the commits in this PR and also https://github.com/OP-TEE/optee_os/pull/1598 you'll notice that certain modification has been done related to the amount of (emulated) SRAM.

**Pull request merged**